lib/enum/arrays.rb

```diff
@@ -1,24 +1,35 @@
 class Array
 
+  def initialize(my_array)
+     @my_array = my_array
+  end
+
   def sum_of_numeric_elements
+     @my_array.select{ |i| i.is_a? (Numeric)}.inject{ |number,sum| number + sum}.to_f
   end
 
   def product_of_numeric_elements
+    @my_array.select{ |i| Integer(i) rescue false}.inject{ |number,mult| number * mult}.to_i.to_f # question about this
```

Comment:
It seems like you're overthinking this one a bit. We need to do two things: first, select the numeric elements, and then multiply them all together. 

To select the numeric elements, you can use `is_a? (Numeric)` like you did in the previous problem. You shouldn't need a rescue. 

Then you can multiply them - same as addition in the previous problem, just use * instead of +. 

The end result: 

`@my_array.select{ |i| i.is_a? Numeric }.inject{ |number, product| number * product}`

And you don't need the conversion to_i or to_f. The result is already a fixnum.